src/schema/v2/me/myCollectionCreateArtworkMutation.ts

```diff
@@ -86,6 +90,19 @@ export const myCollectionCreateArtworkMutation = mutationWithClientMutationId<
         ...rest,
       })
 
+      const artworkId = response.id
+      const regex = /https:\/\/(?<sourceBucket>.*).s3.amazonaws.com\/(?<sourceKey>.*)/
+
+      externalImageUrls = externalImageUrls || []
+      await externalImageUrls.forEach(async (url) => {
```

Comment:
potentially blocking: 

Does this do what we think it does? Iterating over async calls is always challenging, and while this code might "succeed", I don't know if it's doing what we think it's doing. I think it's probably just spinning off the calls to the `myCollectionCreateImageLoader` and moving on, never waiting for any to succeed. 

This article describes the difficulty of iterating over async calls with `forEach`: https://medium.com/dailyjs/the-pitfalls-of-async-await-in-array-loops-cf9cf713bfeb

I usually approach this type of code with `promise.all` (also described in that article).

Reply:
Promise.all will perform all of the requests in parallel too! We should update.

Reply:
What happens when an error occurs in the two awaits?

Reply:
This is now addressed in the tests - the whole thing blows up even though some images were created. 😬 Open to other ideas, but hoping I can convince yall to merge this and then address edge cases in future iterations? 🤞